java/ChartboostMediationDemoJava/src/main/java/com/example/chartboost/mediation/sdk/demo/java/AdController.java

```diff
@@ -0,0 +1,276 @@
+            Button correspondingShowButton,
+            Activity activity,
+            OnLogStateChangeListener onLogStateChangeListener
+    ) {
+        ChartboostMediationFullscreenAd ad = fullscreenAds.get(placementName);
+        if (shouldUseQueue) {
+            ChartboostMediationFullscreenAdQueue queue = getAdQueue(placementName);
+            if (queue.hasNextAd()) {
+                onLogStateChangeListener.logState("Using queue to show the ad");
+                ad = queue.getNextAd();
+                if (ad != null) {
+                    // Be mindful that ads will need to be reattached to your ChartboostMediationFullscreenAdListener if you are listening to ad cycle events.
+                    ad.setListener(createFullscreenAdListener(placementName, correspondingShowButton, onLogStateChangeListener));
+                }
+                fullscreenAds.put(placementName, ad);
+            }
+        }
+        onLogStateChangeListener.logState("Fullscreen ad is about to show");
+        if (ad != null) {
+            ad.showFullscreenAdFromJava(activity, createFullscreenAdShowListener(correspondingShowButton, onLogStateChangeListener));
+        } else {
+            onLogStateChangeListener.logState("Fullscreen ad is null. Load an ad first.");
+            correspondingShowButton.setEnabled(false);
+        }
+    }
+
+    public ChartboostMediationFullscreenAdQueue getAdQueue(String placement) {
+        return adQueues.get(placement);
+    }
+
+    public ChartboostMediationFullscreenAdQueue createAdQueue(
+            String placement,
+            Button correspondingShowButton,
+            Context context,
+            OnLogStateChangeListener onLogStateChangeListener
+    ) {
+        ChartboostMediationFullscreenAdQueue queue = ChartboostMediationFullscreenAdQueueManager.queue(context, placement);
+        queue.setQueueCapacity(46);// this will update queue capacity only to the maximum capacity received on initialization
+        queue.setAdQueueListener(createFullscreenAdQueueListener(correspondingShowButton, onLogStateChangeListener));
+        adQueues.put(placement, queue);
+        return queue;
+    }
+
+    public void setShouldUseQueue(boolean shouldUseQueue) {
+        this.shouldUseQueue = shouldUseQueue;
+    }
+
+    private ChartboostMediationFullscreenAdLoadListener createFullscreenAdLoadListener(
+            String placementName,
+            Button button,
+            OnLogStateChangeListener onLogStateChangeListener
+    ) {
+        return new ChartboostMediationFullscreenAdLoadListener() {
+            @Override
+            public void onAdLoaded(@NonNull ChartboostMediationFullscreenAdLoadResult chartboostMediationFullscreenAdLoadResult) {
+                onLogStateChangeListener.logState("Ad loaded");
+                fullscreenAds.put(placementName, chartboostMediationFullscreenAdLoadResult.getAd());
+                button.setEnabled(fullscreenAds.get(placementName) != null);
```

Comment:
Architecturally, `AdController` shouldn't be interacting with the app UI by touching buttons or layouts directly. That logic should be abstracted to an intermediate layer. It's fine here since it's nothing big, but ideally I'd recommend compartmentalization.